test/gtest/ucp/ucp_test.h

```diff
@@ -19,7 +19,15 @@
 
 #include <queue>
 
-#define MT_TEST_NUM_THREADS       4
+#if _OPENMP
+#include "omp.h"
+#endif
+
+#if _OPENMP && ENABLE_MT
+#define MT_TEST_NUM_THREADS omp_get_max_threads()
```

Comment:
why not omp_get_num_threads () ?

Reply:
https://gcc.gnu.org/onlinedocs/libgomp/omp_005fget_005fnum_005fthreads.html
````
Returns the number of threads in the current team. In a sequential section of the program omp_get_num_threads returns 1.
```
`omp_get_max_threads` returns `OMP_NUM_THREADS` if set or number of cores in a **sequential section**.